Mainnet/AddressMapper/AddressMapper.Tests/AddressMapperTests.cs

```diff
@@ -0,0 +1,145 @@
+using Moq;
+using Stratis.SmartContracts.CLR;
+using Stratis.SmartContracts;
+using Xunit;
+using static AddressMapper;
+
+namespace Tests
+{
+    public class AddressMapperTests
+    {
+        private readonly Mock<ISmartContractState> mContractState;
+        private readonly Mock<IContractLogger> mContractLogger;
+        private readonly Mock<IInternalTransactionExecutor> mInternalExecutor;
+
+        private readonly IPersistentState persistentState;
+
+        private readonly Address contractAddress;
+        private readonly Address primaryAddress;
+        private readonly Address secondaryAddress;
+        private readonly Address ownerAddress;
+
+        public AddressMapperTests()
+        {
+            mContractLogger = new Mock<IContractLogger>();
+            persistentState = new InMemoryState();
+            mContractState = new Mock<ISmartContractState>();
+            mInternalExecutor = new Mock<IInternalTransactionExecutor>();
+            mContractState.Setup(x => x.PersistentState).Returns(persistentState);
+            mContractState.Setup(x => x.ContractLogger).Returns(mContractLogger.Object);
+            mContractState.Setup(x => x.InternalTransactionExecutor).Returns(mInternalExecutor.Object);
+            contractAddress = "0x0000000000000000000000000000000000000001".HexToAddress();
+            ownerAddress = "0x0000000000000000000000000000000000000002".HexToAddress();
+            primaryAddress = "0x0000000000000000000000000000000000000003".HexToAddress();
+            secondaryAddress = "0x0000000000000000000000000000000000000004".HexToAddress();
+        }
+
+        [Theory]
+        [InlineData(Status.Approved)]
+        [InlineData(Status.Pending)]
+        public void MappAddress_Fails_If_SecondaryAddress_Is_Mapped_Already(Status status)
```

Comment:
I don't know the actual use case for these mappings but _should_ there be a way for this to be changed?

Reply:
How ?